js/src/tooltip.js

```diff
@@ -546,7 +553,7 @@ class Tooltip extends BaseComponent {
         EventHandler.on(this._element, eventIn, this.config.selector, event => this._enter(event))
         EventHandler.on(this._element, eventOut, this.config.selector, event => this._leave(event))
       }
-    })
+    }, this)
```

Comment:
Is this really required?

Reply:
![Untitled](https://user-images.githubusercontent.com/22406063/104648172-719e3980-56bb-11eb-9b4a-31eec2c10034.png)


if you remove `this` scope, as you can see on the above debug screenshot, it doesn't see `this` and `this.constructor`  is undefined  . It is not supposed to happen but, as I was debugging I fell on it

Reply:
I am not sure how you are debugging, but this is not required. Because the callback function is an arrow function that gives access to the Tooltip class. :slightly_smiling_face:

Reply:
As i wrote before, I agree with you, but as you can see from the previous and the next screenshot the facts disagree. 

![Untitled](https://user-images.githubusercontent.com/22406063/104774489-70364500-577f-11eb-8213-efe19a5bc2bc.png)


It would be great, if you give it sometime and try it by yourself. 

If you insist to revert it, I am ok with it

Reply:
This shouldn't be needed since we are using an arrow function thus `this` refers to the upper function.

https://stackoverflow.com/questions/19445599/accessing-this-in-a-foreach-loop-results-in-undefined

Maybe we are missing something else?

Reply:
I actually made a commit that is now removed (by force push): https://github.com/twbs/bootstrap/commit/2b02d4949a229325ae5f03c78097ba10b9d74b38

Reply:
Yes, @rohit2sharma95 , we committed the same time, and I missed it. Sorry.